Source/carthage/BuildVersion.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  BuildVersion.swift
+//  Carthage
+//
+//  Created by Rodrigo Garcia on 10/15/16.
+//  Copyright © 2016 Carthage. All rights reserved.
+//
+
+import CarthageKit
+import Commandant
+import Foundation
+import ReactiveCocoa
+import ReactiveTask
+import Result
+import Tentacle
+
+public struct CarthageVersion{
+	
+	public let localVersion:SemanticVersion
+	public let remoteVersion:SemanticVersion
+	
+	public init (localVersion: SemanticVersion, remoteVersion:SemanticVersion){
+		self.localVersion = localVersion
+		self.remoteVersion = remoteVersion
+	}
+	
+	func shouldUpgrade() ->Bool {
+		
+		if localVersion < remoteVersion {
+			return true
+		}
+		
+		return false
+	}
+}
+
+func fetchLatestCarthageVersion() -> CarthageVersion{
```

Comment:
I think it'd be better to split this into two code units:

1. one to get the latest version from the server
2. one to get the local version (which could be used for `carthage version`)